src/deluge/io/midi/device_specific/midi_device_lumi_keys.cpp

```diff
@@ -244,15 +244,26 @@ void MIDIDeviceLumiKeys::setRootNote(int16_t rootNote) {
 // Efficient binary comparison of notes to Lumi builtin scales
 std::pair<MIDIDeviceLumiKeys::Scale, int16_t>
 MIDIDeviceLumiKeys::determineScaleAndRootNoteOffsetFromNotes(uint8_t* modeNotes, uint8_t noteCount) {
+	// First try with all scales "as is" (without transposition)
+	// Turn notes in octave into 12 bit binary
 	uint16_t noteInt = 0;
+	for (uint8_t note = 0; note < noteCount; note++) {
+		noteInt |= 1 << modeNotes[note];
+	}
+	// Compare with Lumis pre-built binary list of scales
+	for (uint8_t scale = 0; scale < MIDI_DEVICE_LUMI_KEYS_SCALE_COUNT; scale++) {
+		if (noteInt == scaleNotes[scale]) {
+			return {Scale(scale), 0};
+		}
+	}
 
-	// Try with all possible transpositions
+	// Then if no matches found, try with all possible transpositions of all the scales
 	for (int32_t i = 0; i < kOctaveSize; i++) {
 		// Turn notes in octave into 12 bit binary
+		noteInt = 0;
 		for (uint8_t note = 0; note < noteCount; note++) {
-			noteInt |= 1 << modeNotes[(note + i) % kOctaveSize];
+			noteInt |= 1 << ((modeNotes[note] + i) % kOctaveSize);
```

Comment:
This line was totally wrong. To be able to correctly transpose a scale, we have to add semitones to the modeNotes[note] value, not accessing a different index from the array as it was before. This was a bug.

Reply:
Ah okay, I get that change now, thank you.